lib/web_ui/lib/src/engine/semantics/label_and_value.dart

```diff
@@ -30,6 +60,42 @@ class LabelAndValue extends RoleManager {
 
   @override
   void update() {
+    final String? computedLabel = _computeLabel();
+
+    if (computedLabel == null) {
+      _cleanUpDom();
+      return;
+    }
+
+    _updateLabel(computedLabel);
+  }
+
+  DomElement? _labelSpan;
+
+  void _updateLabel(String label) {
+    owner.setAttribute('aria-label', label);
```

Comment:
Maybe a comment here explaining why we are setting `aria-label` even when `needsDomText` is true?

Reply:
You made me realize something. I did this originally so that the aria label rolls up to the parent node, which is correctly sized. Span doesn't match the `rect` reported by the framework. However, since I'm only using the span for leaf nodes, I shouldn't need the span at all. I think I can just do `semanticsObject.element.innerText = label`. Let me give it a shot.